lib/google/ads/google_ads/logging_interceptor.rb

```diff
@@ -36,18 +36,31 @@ def request_response(request:, call:, method:, metadata: {})
             customer_id = request.resource_name.split('/').last
           end
           summary_message =
-              sprintf("CID: %s, Host: %s, Method: %s",
-                customer_id,
-                call.instance_variable_get('@wrapped').peer,
-                method
-              )
-          request_message = sprintf("Outgoing request: Headers: %s Payload: %s",
-              metadata.to_json, request.to_json)
+            sprintf("CID: %s, Host: %s, Method: %s",
+                    customer_id,
+                    call.instance_variable_get('@wrapped').peer,
+                    method
+                   )
+
+          # calling #to_json on some protos (specifically those with non-UTF8
+          # encodable byte values) causes a segfault, however #inspect works
+          # so we check if the proto contains a bytevalue, and if it does
+          # we #inspect instead of #to_json
+          request_inspect = if use_bytes_inspect?(request)
```

Comment:
This method as implemented already returns the inspect or to_json call, not true/false. I'm ok with the logic being delegated to a method or being done here, so just pick one and implement that.